include/hnswlib/bruteforce.h

```diff
@@ -5,141 +5,148 @@
+                }
+                else{
+                    if (cur_element_count >= maxelements_) {
+                        throw std::runtime_error("The number of elements exceeds the specified limit\n");
+                    }
+                    idx=cur_element_count;
+                    dict_external_to_internal[label] = idx;
+                    cur_element_count++;
+                }
+            }
+            memcpy(data_ + size_per_element_ * idx + data_size_, &label, sizeof(labeltype));
+            memcpy(data_ + size_per_element_ * idx, datapoint, data_size_);
+
+
+
+
+        };
+
+        void removePoint(labeltype cur_external) {
+            size_t cur_c=dict_external_to_internal[cur_external];
+
+            dict_external_to_internal.erase(cur_external);
+
+            labeltype label=*((labeltype*)(data_ + size_per_element_ * (cur_element_count-1) + data_size_));
+            dict_external_to_internal[label]=cur_c;
+            memcpy(data_ + size_per_element_ * cur_c,
+                   data_ + size_per_element_ * (cur_element_count-1),
+                   data_size_+sizeof(labeltype));
+            cur_element_count--;
+
+        }
+
+
+        std::priority_queue<std::pair<dist_t, labeltype>>
+        searchKnn(const void *query_data, size_t k, size_t batch_index) const {
+            std::priority_queue<std::pair<dist_t, labeltype>> topResults;
+            if (cur_element_count == 0) return topResults;
+            for (int i = 0; i < k; i++) {
+                dist_t dist = fstdistfunc_(query_data, data_ + size_per_element_ * i, dist_func_param_, nullptr);
+                topResults.push(std::pair<dist_t, labeltype>(dist, *((labeltype *)(data_ + size_per_element_ * i +
+                                                                                data_size_))));
+            }
+            dist_t lastdist = topResults.top().first;
+            for (int i = k; i < cur_element_count; i++) {
+                dist_t dist = fstdistfunc_(query_data, data_ + size_per_element_ * i, dist_func_param_, nullptr);
+                if (dist <= lastdist) {
+                    topResults.push(std::pair<dist_t, labeltype>(dist, *((labeltype *) (data_ + size_per_element_ * i +
+                                                                                        data_size_))));
+                    if (topResults.size() > k)
+                        topResults.pop();
+                    lastdist = topResults.top().first;
+                }
+
+            }
+            return topResults;
+        };
+
+        void saveIndex(const std::string &location) {
+            std::ofstream output(location, std::ios::binary);
+            std::streampos position;
+
+            writeBinaryPOD(output, maxelements_);
+            writeBinaryPOD(output, size_per_element_);
+            writeBinaryPOD(output, cur_element_count);
+
+            output.write(data_, maxelements_ * size_per_element_);
+
+            output.close();
```

Comment:
u need to cover with `try` and `catch` errors, or u may have an open stream in bad shape